src/main/java/organice/logic/parser/ExactFindCommandParser.java

```diff
@@ -0,0 +1,45 @@
+package organice.logic.parser;
+
+import static organice.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static organice.logic.parser.CliSyntax.PREFIX_AGE;
+import static organice.logic.parser.CliSyntax.PREFIX_BLOOD_TYPE;
+import static organice.logic.parser.CliSyntax.PREFIX_DOCTOR_IN_CHARGE;
+import static organice.logic.parser.CliSyntax.PREFIX_NAME;
+import static organice.logic.parser.CliSyntax.PREFIX_NRIC;
+import static organice.logic.parser.CliSyntax.PREFIX_ORGAN;
+import static organice.logic.parser.CliSyntax.PREFIX_ORGAN_EXPIRY_DATE;
+import static organice.logic.parser.CliSyntax.PREFIX_PHONE;
+import static organice.logic.parser.CliSyntax.PREFIX_PRIORITY;
+import static organice.logic.parser.CliSyntax.PREFIX_TISSUE_TYPE;
+import static organice.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import organice.logic.commands.ExactFindCommand;
+import organice.logic.parser.exceptions.ParseException;
+import organice.model.person.PersonContainsPrefixesPredicate;
+
+/**
+ * Parses input arguments and creates a new ExactFindCommand object
+ */
+public class ExactFindCommandParser implements Parser<ExactFindCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ExactFindCommand
+     * and returns a ExactFindCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ExactFindCommand parse(String args) throws ParseException {
+        args = args.replaceAll("\n", " ").replaceAll("\\s+", " ");
```

Comment:
What is the purpose of `replaceAll("\n")`? i think trimmed has deal with multi spaces in between right? no need to replace all `\\s+`. Even i think how the tokenizer work also they remove extra spaces. Need to check

Reply:
Replacing newlines is a defensive action on my side of the code. Replacing `\\s+` with `' '` takes care of whitespaces in between individual words (trim only removes preceding and trailing whitespace). Have not tested tokenize, but a glace at the code suggests it doesnt protect against whitespace in between words (I may be wrong). Either way i think its better to be defensive